client/proposer/service.go

```diff
@@ -97,10 +104,29 @@ func (p *Proposer) proposeCollations() {
 				log.Error("Received incorrect p2p message. Wanted a transaction broadcast message")
 				break
 			}
-			// log.Debugf("Received transaction: %x", tx)
-			if err := p.createCollation(p.ctx, []*gethTypes.Transaction{legacyutil.TransformTransaction(tx)}); err != nil {
-				log.Errorf("Create collation failed: %v", err)
+			log.Debugf("Received transaction: %x", tx)
+			gethtx := legacyutil.TransformTransaction(tx)
+			currentperiod, err := p.currentPeriod(p.ctx)
+			if err != nil {
+				log.Errorf("Unable to get current period: %v", err)
+			}
+
+			if (sizeOfCollation+int64(gethtx.Size())) > types.CollationSizeLimit || period.Cmp(currentperiod) != 0 {
```

Comment:
Maybe add a quick comment or two explaining why we're doing this - will be very helpful to someone that has no context of this file in the future